src/main/java/seedu/address/model/person/Person.java

```diff
@@ -178,6 +176,41 @@ public boolean equals(Object other) {
                 && tags.equals(otherPerson.tags);
```

Comment:
Found bug where edit command fails and claims fields are same even though I'm setting a different Telegram field. Solution is to add equals checks for telegram, linkedin, birthday fields here, so that persons are not equal when these fields are different.

Reply:
Should add unit tests for this by the way.

Reply:
Thanks for pointing it out! Didn't really notice on that. Updated the `equals` method in `Person` class accordingly with the relevant test cases as suggested. Fixed in d9bc1b7.